Issue #37 by docwilmot: Add search options for for tags and core/cont…

jenlampton opened the pull request:
Fixes backdrop/backdrop-issues#37
Replaces https://github.com/backdrop/backdrop/pull/1839

quicksketch commented on core/modules/system/js/modules.js:
```diff
@@ -63,20 +50,117 @@ Backdrop.behaviors.moduleFilterByText = {
+      var $sources = $row.find(searchTarget);
+      var coreHidden = $('#edit-core-switch-core').prop("checked") === false;
+      var contribHidden = $('#edit-core-switch-contrib').prop("checked") === false;
+      console.log($sources);
```
Probably needs to be removed

docwilmot commented:
Installed Devel, Honeypot to test.
The core/contrib switch isnt working for me, unless I first select a tag from the dropdown.

jenlampton commented:
@docwilmot I think I have both the core/contrib filter and the tags selector working now. Closing for a new sandbox.

jenlampton commented:
sandbox!

jenlampton commented:
Closing for a new sandbox.

jenlampton commented:
Sandbox!

jenlampton commented:
Try a new sandbox?

jenlampton commented:
sandbox me?

docwilmot commented:
This seems to be broken. @jenlampton is this working locally for you? Getting a 500 error with most admin pages

docwilmot commented:
There are a few thousand errors about `"core" is an invalid render array key`; suspect its this:
```
  // Add core/contrib filter.
  $form['filter']['source'] = array(
    '#type' => 'checkboxes',
    '#title' => t('Module source'),
    '#default_value' => array('core', 'contrib'),
    '#options' => array(
      'core' => 'Backdrop core',
      'contrib' => 'Contributed projects',
    ),
    'core' => array('#attributes' => array('data-filter-source' => 'core')),
    'contrib' => array('#attributes' => array('data-filter-source' => 'contrib')),
  );
```
Perhaps this is killing the sandbox.

quicksketch commented on core/themes/seven/css/style.css:
```diff
@@ -777,8 +777,15 @@ div.teaser-checkbox .form-item,
 .form-item label {
-  margin: 0 0 10px;
+  margin: 0 0 5px;
+}
```
These styling changes seem like they would affect every form in the admin area. Are these necessary?

jenlampton replied:
Necessary no, no CSS changes are ever necessary. But this does fix form-element styling across all of Seven, which was a little out of whack. I'm happy to do that in a separate issue, but the only place in core that we can see it's so funky is on this page (and for views exposed filters, but those are already overridden & fixed there)

jenlampton commented:
> This seems to be broken. @jenlampton is this working locally for you?

Yes, it's still working locally.

> suspect its this:

Hm, that's the [syntax](https://www.drupal.org/project/drupal/issues/2779999) for adding attributes to form element options... And I'm not seeing those errors. 

edit: Aha! I can see them now..  pushing a new branch w/ slightly different syntax.

jenlampton commented:
closing again for a new sandbox.

docwilmot commented:
oops sorry if youre still working on this, reopened.

docwilmot commented on core/modules/system/system.admin.inc:
```diff
@@ -518,13 +518,63 @@ function system_modules($form, $form_state = array()) {
   foreach ($visible_files as $filename => $file) {
-    if (!empty($file->info['hidden'])) {
+    // Remove hidden and required modules from display list.
+    if (!empty($file->info['hidden'] || !empty($file->info['required']))) {
```
Missing end bracket for first `empty()`

jenlampton commented:
Thanks @docwilmot and @Gormartsen for you sleuthing!  The problem was that I had converted `#core` to `core` to prevent it from looking so much like a form API property, but since the value was a string instead of an array, Backdrop couldn't handle it. 

@kolons -- you're going to love this -- since I had to change that property anyway, I took some code from your PR to make it into `#source` instead. Now it contains either `core` or `contrib` and I also added support for `custom`.

jenlampton commented:
closing for a new sandbox.

jenlampton commented:
sandboxme!
